pkg/datastore/datastore.go

```diff
@@ -188,8 +188,8 @@ type ReadWriteTransaction interface {
 	// WriteNamespaces takes proto namespace definitions and persists them.
 	WriteNamespaces(newConfigs ...*core.NamespaceDefinition) error
 
-	// DeleteNamespace deletes a namespace and any associated tuples.
-	DeleteNamespace(nsName string) error
+	// DeleteNamespaces deletes namespaces including associated Relationships.
```

Comment:
TIL `DeleteNamespaces` also deletes relationships! It would be quite handy to expose this as an option in the `WriteSchema` API so that folks don't have to write client code that does the deletion of the relationships! @josephschorr ?

Reply:
Folks having to explicitly empty namespaces is an enforced requirement on purpose: it ensures that someone cannot *accidentally* delete data by simply removing a definition from a schema. The deletion of relationships within the DeleteNamespace call is just an extra check to ensure that we are not left with dangling data

Reply:
yeah, I was just thinking of some sort of "do only if you know what you are doing" kind of flag. But yeah, I understand safety and doing the right thing by default comes first